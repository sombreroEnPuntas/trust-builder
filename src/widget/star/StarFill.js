import React from 'react';
import PropTypes from 'prop-types';

import Star from './Star';
import './StarFill.scss';

const StarFill = ({ rating }) => {
  let line = [];
  for (let i =1;i <= 5; i++){
    line.push(<Star key={`star-color-${rating}-line-${i}`} index={i} rating={rating} />);
  }
  
  return <span className="StarFill">{line}</span>;
};

StarFill.propTypes = {
  rating: PropTypes.number.isRequired,
};

export default StarFill;
