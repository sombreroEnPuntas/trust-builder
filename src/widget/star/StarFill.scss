@import '../styles';

.StarFill {
  &-floating {
    @include display-flex-center();

    .positioning {
      display: inline-block;
    }

    .positioning:nth-child(1) {
      z-index: 0;
      transform: rotate(40deg) scale(0.7) translate(60%, -120%);
    }
    .positioning:nth-child(2) {
      z-index: 1;
      transform: rotate(20deg) scale(0.9) translate(35%, -25%);
    }
    .positioning:nth-child(3) {
      z-index: 2;
    }
    .positioning:nth-last-child(2) {
      z-index: 1;
      transform: rotate(-20deg) scale(0.9) translate(-35%,-25%);
    }
    .positioning:nth-last-child(1) {
      z-index: 0;
      transform: rotate(-40deg) scale(0.7) translate(-60%, -120%);
    }
  }
}
