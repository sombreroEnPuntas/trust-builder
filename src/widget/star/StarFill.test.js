import React from 'react';
import { shallow } from 'enzyme';

import StarFill from './StarFill';

it('StarFill renders without crashing', () => {
  shallow(<StarFill rating={5} />);
});

describe('StarFill', () => {
  const ratings = [
    {rating: 1}, {rating: 2}, {rating: 3}, {rating: 4}, {rating: 5},
  ];

  ratings.forEach(rating => {
    it(`for rating: ${rating.rating}`, () => {
      const wrapper = shallow(<StarFill {...rating} />);
      expect(wrapper.debug()).toMatchSnapshot();
    });
  });

  it('has floating class', () => {
    const wrapper = shallow(<StarFill rating={5} floating />);
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
