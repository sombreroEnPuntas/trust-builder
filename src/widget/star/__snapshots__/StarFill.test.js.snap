// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StarFill for rating: 1 1`] = `
"<span className=\\"StarFill\\">
  <Star index={1} rating={1} size=\\"small\\" />
  <Star index={2} rating={1} size=\\"small\\" />
  <Star index={3} rating={1} size=\\"small\\" />
  <Star index={4} rating={1} size=\\"small\\" />
  <Star index={5} rating={1} size=\\"small\\" />
</span>"
`;

exports[`StarFill for rating: 2 1`] = `
"<span className=\\"StarFill\\">
  <Star index={1} rating={2} size=\\"small\\" />
  <Star index={2} rating={2} size=\\"small\\" />
  <Star index={3} rating={2} size=\\"small\\" />
  <Star index={4} rating={2} size=\\"small\\" />
  <Star index={5} rating={2} size=\\"small\\" />
</span>"
`;

exports[`StarFill for rating: 3 1`] = `
"<span className=\\"StarFill\\">
  <Star index={1} rating={3} size=\\"small\\" />
  <Star index={2} rating={3} size=\\"small\\" />
  <Star index={3} rating={3} size=\\"small\\" />
  <Star index={4} rating={3} size=\\"small\\" />
  <Star index={5} rating={3} size=\\"small\\" />
</span>"
`;

exports[`StarFill for rating: 4 1`] = `
"<span className=\\"StarFill\\">
  <Star index={1} rating={4} size=\\"small\\" />
  <Star index={2} rating={4} size=\\"small\\" />
  <Star index={3} rating={4} size=\\"small\\" />
  <Star index={4} rating={4} size=\\"small\\" />
  <Star index={5} rating={4} size=\\"small\\" />
</span>"
`;

exports[`StarFill for rating: 5 1`] = `
"<span className=\\"StarFill\\">
  <Star index={1} rating={5} size=\\"small\\" />
  <Star index={2} rating={5} size=\\"small\\" />
  <Star index={3} rating={5} size=\\"small\\" />
  <Star index={4} rating={5} size=\\"small\\" />
  <Star index={5} rating={5} size=\\"small\\" />
</span>"
`;
